x-pack/plugins/alerting/server/routes/health.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  IRouter,
+  RequestHandlerContext,
+  KibanaRequest,
+  IKibanaResponse,
+  KibanaResponseFactory,
+  ElasticsearchServiceSetup,
+} from 'kibana/server';
+import { AlertingFrameworkHealth } from '../types';
+
+interface XPackUsageSecurity {
+  security?: {
+    enabled?: boolean;
+    ssl?: {
+      http?: {
+        enabled?: boolean;
+      };
+    };
+  };
+}
+
+export function healthRoute(router: IRouter, elasticsearch: ElasticsearchServiceSetup) {
+  router.get(
+    {
+      path: '/api/alert/_health',
+      validate: false,
+    },
+    router.handleLegacyErrors(async function(
+      context: RequestHandlerContext,
+      req: KibanaRequest<any, any, any, any>,
+      res: KibanaResponseFactory
+    ): Promise<IKibanaResponse<any>> {
+      try {
+        const {
+          security: {
+            enabled: isSecurityEnabled = false,
+            ssl: { http: { enabled: isTLSEnabled = false } = {} } = {},
+          } = {},
+        } = await elasticsearch.adminClient
+          // `transport.request` is potentially unsafe when combined with untrusted user input.
+          // Do not augment with such input.
+          .callAsInternalUser<XPackUsageSecurity>('transport.request', {
+            method: 'GET',
+            path: '/_xpack/usage',
+          });
+
+        const frameworkHealth: AlertingFrameworkHealth = {
+          canGenerateApiKeys: !isSecurityEnabled || (isSecurityEnabled && isTLSEnabled),
```

Comment:
Oh, I thought we said you could if security was disabled and if security was enabled on TLS.

So that mean we don't require keys for alerting to work when security is disabled?

What do you thin ka good name for this would be then? 🤔

Reply:
Added the license check, but I'm unsure it really should be here 🤔

Reply:
> Oh, I thought we said you could if security was disabled and if security was enabled on TLS.

The concept of users in Kibana is tied to security being enabled so that goes away when security is disabled (you'll notice the login page goes away too).

> So that mean we don't require keys for alerting to work when security is disabled?

That's correct

> What do you think a good name for this would be then?

I'm not the best at coming up with names but I'll throw a few: `requiresTLSSetup`, `mustEnableTLS` or 2 variables `securityEnabled` & `tlsEnabled`.

> Added the license check, but I'm unsure it really should be here

I think for consistency this is fine. I'm not sure how running OSS license works but I think we would have to pretend our plugin is disabled dynamically.